tests/integration/connection.rs

```diff
@@ -1,85 +1,88 @@
+use async_trait::async_trait;
 use datastore_entity::connection::DatastoreConnection;
-use gcp_auth::{Error as GCPAuthError, Token};
-use google_datastore1::Client;
+use gcp_auth::{AuthenticationManager, Error as GCPAuthError};
 use google_api_auth::GetAccessToken;
+use google_datastore1::Client;
 
-use std::error::Error;
 use std::env;
-
+use std::error::Error;
+use std::fmt::{Debug, Formatter};
 
 //
-// Google Access Token
+// An opaque newtype type that can generate access tokens
 //
-#[derive(Debug)]
-pub(crate) struct GoogleToken(pub Token);
+pub(crate) struct GoogleAuthentication(AuthenticationManager);
```

Comment:
This is a better design. Previously it could only use the access token until it expired, now it will fetch new ones as needed.

Reply:
👍